File: suc/kotlin/021.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "021KanbanBoarden"
private const val UI_TYPE = "Productivity"
private const val STYLE_THEME = "Modern Blue"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFF2563EB)
        val secondary = Color(0xFF60A5FA)
        val tertiary = Color(0xFF3B82F6)
        val background = Color(0xFFF3F4F6)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFE5E7EB)
        val outline = Color(0xFFD1D5DB)
        val success = Color(0xFF16A34A)
        val warning = Color(0xFFF59E0B)
        val error = Color(0xFFDC2626)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFF1E1E1E)
        val onTertiary = Color(0xFFFFFFFF)
        val onBackground = Color(0xFF1E1E1E)
        val onSurface = Color(0xFF1E1E1E)
    }
    object TypographyTokens {
        val display = TextStyle(fontSize = 28.sp, fontWeight = FontWeight.Bold)
        val title = TextStyle(fontSize = 18.sp, fontWeight = FontWeight.Medium)
        val body = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Normal)
        val label = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Medium)
    }
    object Shapes {
        val small = RoundedCornerShape(6.dp)
        val medium = RoundedCornerShape(10.dp)
        val large = RoundedCornerShape(14.dp)
    }
    object Spacing {
        val xs = 4.dp
        val sm = 8.dp
        val md = 12.dp
        val lg = 16.dp
        val xl = 24.dp
    }
    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(2.dp, 4.dp, 2.dp, 0.12f)
        val level2 = ShadowSpec(4.dp, 8.dp, 4.dp, 0.16f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large
        ),
        content = content
    )
}

data class KanbanColumn(val title: String, val tasks: List<String>)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    val columns = listOf(
        KanbanColumn("To Do", listOf("Write report", "Design mockup")),
        KanbanColumn("In Progress", listOf("Implement feature X", "Fix layout bug")),
        KanbanColumn("Done", listOf("Team sync", "Client feedback"))
    )
    var notificationsEnabled by remember { mutableStateOf(true) }
    Scaffold(
        contentWindowInsets = WindowInsets(0),
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text("Kanban Board", style = MaterialTheme.typography.displayLarge, color = AppTokens.Colors.onSurface) },
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors(containerColor = AppTokens.Colors.background)
            )
        },
        containerColor = AppTokens.Colors.background
    ) { pad ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(pad)
                .padding(AppTokens.Spacing.lg),
            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg)
        ) {
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween,
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text("Notifications", style = MaterialTheme.typography.titleMedium, color = AppTokens.Colors.onSurface)
                Switch(
                    checked = notificationsEnabled,
                    onCheckedChange = { notificationsEnabled = it },
                    colors = SwitchDefaults.colors(checkedThumbColor = AppTokens.Colors.primary)
                )
            }
            LazyRow(
                horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md),
                modifier = Modifier.fillMaxWidth()
            ) {
                items(columns) { col ->
                    Card(
                        shape = AppTokens.Shapes.large,
                        colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.surface),
                        elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level2.elevation),
                        modifier = Modifier
                            .width(220.dp)
                            .fillMaxHeight()
                    ) {
                        Column(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(AppTokens.Spacing.md),
                            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)
                        ) {
                            Text(col.title, style = MaterialTheme.typography.titleMedium, color = AppTokens.Colors.primary)
                            col.tasks.forEach { task ->
                                AssistChip(
                                    onClick = {},
                                    label = { Text(task, style = MaterialTheme.typography.bodyMedium, color = AppTokens.Colors.onSurface) },
                                    colors = AssistChipDefaults.assistChipColors(containerColor = AppTokens.Colors.surfaceVariant),
                                    shape = AppTokens.Shapes.medium
                                )
                            }
                        }
                    }
                }
            }
            Text("Task Completion", style = MaterialTheme.typography.titleMedium, color = AppTokens.Colors.onSurface)
            LinearProgressIndicator(
                progress = { 0.65f },
                modifier = Modifier.fillMaxWidth().height(8.dp),
                color = AppTokens.Colors.primary,
                trackColor = AppTokens.Colors.surfaceVariant
            )
        }
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFF3F4F6)
@Composable
fun PreviewScreen() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/021.swift
import SwiftUI
import UIKit // Required for UIHostingController and prefersStatusBarHidden

// MARK: - AppTokens
// 严格遵守要求4：按钮尺寸、间距与留白需与安卓版本原子级对应，保持控件样式、比例及排版一致。
// 所有 Dp 值直接转换为 CGFloat。
// 所有 Color 值从十六进制直接转换为 RGB。
// 所有 Font 尺寸和权重直接翻译。
struct AppTokens {
    struct Colors {
        static let primary = Color(red: 0x25 / 255.0, green: 0x63 / 255.0, blue: 0xEB / 255.0)
        static let secondary = Color(red: 0x60 / 255.0, green: 0xA5 / 255.0, blue: 0xFA / 255.0)
        static let tertiary = Color(red: 0x3B / 255.0, green: 0x82 / 255.0, blue: 0xF6 / 255.0)
        static let background = Color(red: 0xF3 / 255.0, green: 0xF4 / 255.0, blue: 0xF6 / 255.0)
        static let surface = Color(red: 0xFF / 255.0, green: 0xFF / 255.0, blue: 0xFF / 255.0)
        static let surfaceVariant = Color(red: 0xE5 / 255.0, green: 0xE7 / 255.0, blue: 0xEB / 255.0)
        static let outline = Color(red: 0xD1 / 255.0, green: 0xD5 / 255.0, blue: 0xDB / 255.0)
        static let success = Color(red: 0x16 / 255.0, green: 0xA3 / 255.0, blue: 0x4A / 255.0)
        static let warning = Color(red: 0xF5 / 255.0, green: 0x9E / 255.0, blue: 0x0B / 255.0)
        static let error = Color(red: 0xDC / 255.0, green: 0x26 / 255.0, blue: 0x26 / 255.0)
        static let onPrimary = Color(red: 0xFF / 255.0, green: 0xFF / 255.0, blue: 0xFF / 255.0)
        static let onSecondary = Color(red: 0x1E / 255.0, green: 0x1E / 255.0, blue: 0x1E / 255.0)
        static let onTertiary = Color(red: 0xFF / 255.0, green: 0xFF / 255.0, blue: 0xFF / 255.0)
        static let onBackground = Color(red: 0x1E / 255.0, green: 0x1E / 255.0, blue: 0x1E / 255.0)
        static let onSurface = Color(red: 0x1E / 255.0, green: 0x1E / 255.0, blue: 0x1E / 255.0)
    }

    struct TypographyTokens {
        static let display = Font.system(size: 28, weight: .bold)
        static let title = Font.system(size: 18, weight: .medium)
        static let body = Font.system(size: 14, weight: .regular) // FontWeight.Normal 映射到 .regular
        static let label = Font.system(size: 12, weight: .medium)
    }

    struct Shapes {
        let cornerRadius: CGFloat
        static let small = Shapes(cornerRadius: 6)
        static let medium = Shapes(cornerRadius: 10)
        static let large = Shapes(cornerRadius: 14)
    }

    struct Spacing {
        static let xs: CGFloat = 4
        static let sm: CGFloat = 8
        static let md: CGFloat = 12
        static let lg: CGFloat = 16
        static let xl: CGFloat = 24
    }

    struct ShadowSpec {
        let elevation: CGFloat
        let radius: CGFloat
        let dy: CGFloat
        let opacity: Double
    }

    struct ElevationMapping {
        // Kotlin: ShadowSpec(2.dp, 4.dp, 2.dp, 0.12f) -> radius: 4, y: 2, opacity: 0.12
        static let level1 = ShadowSpec(elevation: 2, radius: 4, dy: 2, opacity: 0.12)
        // Kotlin: ShadowSpec(4.dp, 8.dp, 4.dp, 0.16f) -> radius: 8, y: 4, opacity: 0.16
        static let level2 = ShadowSpec(elevation: 4, radius: 8, dy: 4, opacity: 0.16)
    }
}

// MARK: - Data Models
struct KanbanColumn: Identifiable {
    let id = UUID() // SwiftUI ForEach 需要 Identifiable
    let title: String
    let tasks: [String]
}

// MARK: - RootScreen
struct RootScreen: View {
    let columns = [
        KanbanColumn(title: "To Do", tasks: ["Write report", "Design mockup"]),
        KanbanColumn(title: "In Progress", tasks: ["Implement feature X", "Fix layout bug"]),
        KanbanColumn(title: "Done", tasks: ["Team sync", "Client feedback"])
    ]

    @State private var notificationsEnabled: Bool = true

    var body: some View {
        // 要求2: App 必须设置为全屏显示，并确保顶部状态栏隐藏。
        // 全屏背景色，忽略所有安全区域。
        ZStack {
            AppTokens.Colors.background.ignoresSafeArea(.all)

            VStack(spacing: 0) { // 主内容 VStack，内部元素自行处理间距
                // 自定义顶部栏，等同于 CenterAlignedTopAppBar
                VStack(spacing: 0) {
                    HStack {
                        Spacer()
                        Text("Kanban Board")
                            .font(AppTokens.TypographyTokens.display)
                            .foregroundColor(AppTokens.Colors.onSurface)
                        Spacer()
                    }
                    .padding(.horizontal, AppTokens.Spacing.lg) // 水平内边距
                    .padding(.top, AppTokens.Spacing.lg) // 顶部内边距
                    .padding(.bottom, AppTokens.Spacing.md) // 底部内边距
                }
                .frame(maxWidth: .infinity)
                .background(AppTokens.Colors.background) // 顶部栏背景色

                // 主要内容区域
                VStack(alignment: .leading, spacing: AppTokens.Spacing.lg) {
                    // 通知行
                    HStack {
                        Text("Notifications")
                            .font(AppTokens.TypographyTokens.title)
                            .foregroundColor(AppTokens.Colors.onSurface)
                        Spacer()
                        Toggle(isOn: $notificationsEnabled) {
                            // 空标签，因为文本已单独提供
                        }
                        .toggleStyle(SwitchToggleStyle(tint: AppTokens.Colors.primary)) // 匹配 checkedThumbColor
                        .labelsHidden() // 隐藏默认的 Toggle 标签
                    }
                    .frame(maxWidth: .infinity)

                    // Kanban 列的 LazyRow 等效
                    ScrollView(.horizontal, showsIndicators: false) {
                        LazyHStack(spacing: AppTokens.Spacing.md) {
                            ForEach(columns) { col in
                                CardView(column: col)
                                    .frame(width: 220) // 固定宽度，与 Kotlin 相同
                                    .frame(maxHeight: .infinity) // 填充高度，与 Kotlin 相同
                            }
                        }
                        .padding(.vertical, AppTokens.Spacing.xs) // 滚动视图内容的小垂直内边距
                    }

                    // 任务完成
                    Text("Task Completion")
                        .font(AppTokens.TypographyTokens.title)
                        .foregroundColor(AppTokens.Colors.onSurface)

                    // LinearProgressIndicator 等效
                    ProgressView(value: 0.65)
                        .progressViewStyle(LinearProgressViewStyle(tint: AppTokens.Colors.primary, trackColor: AppTokens.Colors.surfaceVariant))
                        .frame(maxWidth: .infinity)
                        .frame(height: 8) // 固定高度，与 Kotlin 相同
                }
                .padding(AppTokens.Spacing.lg) // 主要内容 VStack 的内边距
                .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .topLeading) // 填充剩余空间
            }
        }
    }
}

// MARK: - Custom ProgressViewStyle to match LinearProgressIndicator
struct LinearProgressViewStyle: ProgressViewStyle {
    var tint: Color
    var trackColor: Color

    func makeBody(configuration: Configuration) -> some View {
        GeometryReader { geometry in
            ZStack(alignment: .leading) {
                RoundedRectangle(cornerRadius: 4) // 高度 8.dp 的一半
                    .fill(trackColor)
                    .frame(height: 8)

                RoundedRectangle(cornerRadius: 4)
                    .fill(tint)
                    .frame(width: geometry.size.width * CGFloat(configuration.fractionCompleted ?? 0), height: 8)
            }
        }
    }
}

// MARK: - CardView (Kanban Column Card)
struct CardView: View {
    let column: KanbanColumn

    var body: some View {
        VStack(alignment: .leading, spacing: AppTokens.Spacing.sm) {
            Text(column.title)
                .font(AppTokens.TypographyTokens.title)
                .foregroundColor(AppTokens.Colors.primary)

            ForEach(column.tasks, id: \.self) { task in
                AssistChipView(task: task)
            }
            Spacer() // 将内容推到卡片顶部，模拟 fillMaxHeight 行为
        }
        .padding(AppTokens.Spacing.md)
        .background(AppTokens.Colors.surface)
        .cornerRadius(AppTokens.Shapes.large.cornerRadius)
        .shadow(color: Color.black.opacity(AppTokens.ElevationMapping.level2.opacity),
                radius: AppTokens.ElevationMapping.level2.radius,
                x: 0,
                y: AppTokens.ElevationMapping.level2.dy)
    }
}

// MARK: - AssistChipView
struct AssistChipView: View {
    let task: String

    var body: some View {
        Button(action: {
            // Kotlin 版本中 onClick 为空，此处也留空
        }) {
            Text(task)
                .font(AppTokens.TypographyTokens.body)
                .foregroundColor(AppTokens.Colors.onSurface)
                .padding(.horizontal, AppTokens.Spacing.lg) // 调整内边距以匹配 AssistChip 视觉效果
                .padding(.vertical, AppTokens.Spacing.sm) // 调整垂直内边距
                .background(AppTokens.Colors.surfaceVariant)
                .cornerRadius(AppTokens.Shapes.medium.cornerRadius)
        }
        .buttonStyle(PlainButtonStyle()) // 移除默认按钮样式
    }
}

// MARK: - UIHostingController for Status Bar Hiding (要求2)
// 必须在 Info.plist 中设置 "View controller-based status bar appearance" 为 YES。
class HostingController<Content>: UIHostingController<Content> where Content : View {
    override var prefersStatusBarHidden: Bool {
        return true // 隐藏顶部状态栏
    }
    
    // 确保在 iOS 16.0 及以上版本兼容
    override var preferredStatusBarUpdateAnimation: UIStatusBarAnimation {
        return .fade
    }
}

// MARK: - App Entry Point (要求5)
@main
struct KanbanBoardApp: App {
    var body: some Scene {
        WindowGroup {
            // 将 RootScreen 包装在自定义的 HostingController 中以隐藏状态栏。
            HostingControllerWrapper()
        }
    }
}

// Helper struct to wrap UIHostingController in a SwiftUI View
struct HostingControllerWrapper: UIViewControllerRepresentable {
    func makeUIViewController(context: Context) -> HostingController<RootScreen> {
        return HostingController(rootView: RootScreen())
    }

    func updateUIViewController(_ uiViewController: HostingController<RootScreen>, context: Context) {
        // SwiftUI 视图更新时，无需在此处执行任何操作
    }
}